File: Data Structures & Algorithms/remove-linked-list-elements/submission-3.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func removeElements(head *ListNode, val int) *ListNode {
    sentinel := ListNode{
        Val: 0,
        Next: nil,
    }
    sentinel.Next = head;

    prev := &sentinel;

    for prev != nil  && prev.Next != nil {
        if prev.Next.Val == val {
            prev.Next = prev.Next.Next;
        } else {
            prev = prev.Next;
        }
    }
    return sentinel.Next;
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-3.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        sentinel = ListNode()
        sentinel.next = head
        length = 0
        while head:
            length += 1
            head = head.next

        prevPos = length - n
        prev = sentinel
        for _ in range(prevPos):
            prev = prev.next

        prev.next = prev.next.next
        return sentinel.next
        